packages/studio/src/components/PillPicker/PillPicker.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useRef, useState } from "react";
+import { ReactComponent as EmbedIcon } from "../../icons/embed.svg";
+import { useRootClose } from "@restart/ui";
+import classNames from "classnames";
+
+interface PillPickerProps {
+  availableItems: string[];
+  selectItem: (item: string) => void;
+  disabled?: boolean;
+  getDisplayName?: (item: string) => string;
+}
+
+/**
+ * An icon that displays a dropdown of available items when clicked.
+ */
+export default function PillPicker({
+  availableItems,
+  selectItem,
+  disabled,
+  getDisplayName,
+}: PillPickerProps) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useRootClose(containerRef, () => {
+    setIsOpen(false);
+  });
+
+  const toggleOpen = useCallback(() => {
+    if (disabled) {
+      return;
+    }
+    setIsOpen((isOpen) => !isOpen);
+  }, [disabled]);
+
+  const embedIconClasses = classNames("opacity-50", {
+    "cursor-pointer hover:opacity-100": !disabled,
+    "cursor-default": disabled,
+  });
+
+  return (
+    <div className="ml-auto mb-1" ref={containerRef}>
+      <EmbedIcon
+        role="button"
+        onClick={toggleOpen}
+        className={embedIconClasses}
+        aria-label="Toggle pill picker"
+      />
+      {isOpen && renderDropdown(availableItems, selectItem, getDisplayName)}
+    </div>
+  );
+}
+
+function renderDropdown(
```

Comment:
Maybe not a right-now issue, but would it be useful to have this dropdown and the field picker dropdown make use of a shared dropdown component?

Reply:
Right now, this dropdown is much simpler than the field picker dropdown (which has recursive, nested dropdowns). But, in SLAP-2907, we're going to need a more complex dropdown for entities, so it may make sense to make a shared dropdown as part of that sub-task. I'll call that out in its description!